### PR TITLE
USB MIDI 2.0 Group Terminal Blocks

### PR DESCRIPTION
**Pull request by AmeNote-Michael**
Fetch and parse group terminal blocks for USB MIDI 2.0 device. Will create data in new GTB memory structure as KSMULTIPLE_ITEMS and make available as a property.

### CODE REVIEW

**Comment by AmeNote-Michael** [comment]
Added further code cleanup and removal of unused elements.

File: src/usb-driver/USBMIDI2/Device.cpp [hunk 1]
```diff
@@ -2211,7 +2095,7 @@ Return Value:Amy
             if (!NT_SUCCESS(status))
             {
                 TraceEvents(TRACE_LEVEL_ERROR, TRACE_DEVICE,
-                    "%!FUNC! could not create WdfMemory with status: 0x%x.\n", status);
+                    "%!FUNC! could not create WdfMemory with status: %!STATUS!", status);
                 goto DriverIoWriteExit;
             }
```
Comment (garydan42) [inline src/usb-driver/USBMIDI2/Device.cpp:2101]: signed/unsigned mismatch on 2104

File: src/usb-driver/USBMIDI2/ump.h [hunk 2]
```diff
@@ -335,19 +335,11 @@ typedef struct
...
-
 /// MIDI 2 Group Terminal Blocks Descriptor with multiple terminal blocks
-#define midi2_cs_interface_desc_group_terminal_blocks_n_t(group_terminal_block_num) \
-  struct {       \
-    midi2_desc_group_terminal_block_header_t header;            \
-    midi2_desc_group_terminal_block_t aBlock[group_terminal_block_num]; \
-  }
+typedef struct {
+    midi2_desc_group_terminal_block_header_t header;
+    midi2_desc_group_terminal_block_t aBlock[];
```
[the remark this answers is not in the document]
Comment (garydan42) [reply src/usb-driver/USBMIDI2/ump.h:341]: This hasn't been addressed in the latest push.

[the remark this answers is not in the document]
Comment (AmeNote-Michael) [reply src/usb-driver/USBMIDI2/ump.h]: Now resolved.

**Review by garydan42 — APPROVED** [review]

**Pull request merged**